src/fft_2n.h

```diff
@@ -297,33 +304,51 @@ void Radix2<T>::fft(vec::Buffers<T>& output, vec::Buffers<T>& input)
 }
 
 /** Perform decimation-in-frequency FFT or inverse FFT
+ * Input buffer is in reversed-bit order. Hence butterfly operations can
+ * be performed directly on this order. But the output must be re-order in the
+ * natural order by using swap operations. A swap operation is to swap only two
+ * buffers' pointer, hence it's not costly. However, it will modify the output
+ * buffers. To avoid that, we shuffle the output twice, before and after the
```

Comment:
It's not really a shuffle, you just reverse it. (shuffling is usually used in term of random shuffling).